### PR TITLE
[GLT-3827] Upgrade to Java 17

### PR DESCRIPTION
**Pull request by avarsava**
JIRA Ticket: GLT-3827

- [X] Updates release notes
- [X] Updates developer documentation

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ Full instructions for the setup and use of Run Scanner can be found in the [Run
 
 ## Prerequisites
 
-* JDK 11
+* JDK 17
 * Tomcat 8 or 9
 * C++ build environment (see [runscanner-llumina](runscanner-illumina/README.md))
 * [Maven 3.0.5](http://maven.apache.org/download.html) or later
```
Comment (avarsava) [inline README.md:14]: Oh, a Maven version this old will probably not play nice with Java 17? I am using 3.8.5, but is that OK to make the minimum or is there a version earlier than that I should specify?

Comment (djcooke) [reply README.md:14]: I think Maven 3.8+ is good

**Comment by djcooke** [comment]
GitHub CI workflow needs updated

File: README.md (hunk 1 above)
Comment (avarsava) [reply README.md]: Updated, along with CI

**Comment by avarsava** [comment]
CI passed :relieved:

**Review by djcooke — APPROVED** [review]

**Review by callunity — APPROVED** [review]

**Review by djcooke — APPROVED** [review]

**Pull request merged**